processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -244,6 +249,7 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             Timer timer = Utils.timer();
             contexts.destroyAllProcessors();
             try {
+                contexts.updateHighWatermarks();
```

Comment:
Any points to put this in inside try block while we don't expect it to throw?

Reply:
Let's leave a comment describing this situation more like:
"Since kafka version xx the rebalance listener gets triggered at close so the same thing is supposed to happen, but its not true for older kafka-clients version so we're manually handling this here instead of relying on rebalance listener for better compatibility" something like this.

Reply:
Hm, no.
I'll move to outside of try block.
> Any points to put this in inside try block